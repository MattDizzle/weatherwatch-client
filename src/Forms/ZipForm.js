import React, { Component } from 'react';
import { CurrentContext } from '../CurrentContext';
import ValidationError from '../ValidationError';

export default class ZipForm extends Component {
    static contextType = CurrentContext;

    validateZipcode() {
      const zipcode = parseInt(this.context.zipcode.value);
      if (zipcode.toString().length < 5) {
        return "Zip code must be at least 5 numbers long";
      } 
    }

    render() {
        const { getDataByZip, updateZipcode} = this.context;
        const { zipcode } = this.context
        const zipError = this.validateZipcode();
        return (
          <form onSubmit={getDataByZip}>
            <label htmlFor="zip">Enter a Zip Code </label>
            <input
              id="zip"
              name="zip"
              type="text"
              pattern="[0-9]*"
              required
              onChange={(e) => updateZipcode(e.target.value)}
            />
            {zipcode.touched && (
                <ValidationError message={zipError} />
              )}
            
            <button
              type="submit"
              className="btn"
              disabled={this.validateZipcode()}
            >
              <span>Submit</span>
            </button>
          </form>
        )
    }
}
