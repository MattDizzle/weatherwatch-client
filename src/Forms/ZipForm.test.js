import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import ZipForm from "./ZipForm";
import CurrentContextProvider from "../CurrentContext";


describe(`ZipForm Component`, () => {
    describe(`Smoke test`, () => {
        it("should render without crashing", () => {
            const div = document.createElement('div');
            ReactDOM.render(
            <CurrentContextProvider>
            <ZipForm/>
            </CurrentContextProvider>,
             div);
            ReactDOM.unmountComponentAtNode(div);
          });
    })

    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
          const div = document.createElement('div');
            const tree = renderer.create(
            <CurrentContextProvider>
            <ZipForm/>
            </CurrentContextProvider>
            ).toJSON()
            expect(tree).toMatchSnapshot()
        });
    });

})


