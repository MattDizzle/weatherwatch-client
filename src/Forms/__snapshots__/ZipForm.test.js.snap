// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipForm Component Snapshot test Renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="zip"
  >
    Enter a Zip Code 
  </label>
  <input
    id="zip"
    name="zip"
    onChange={[Function]}
    pattern="[0-9]*"
    required={true}
    type="text"
  />
  <button
    className="btn"
    disabled="Zip code must be at least 5 numbers long"
    type="submit"
  >
    <span>
      Submit
    </span>
  </button>
</form>
`;
